chain/network/src/edge_verifier.rs

```diff
@@ -0,0 +1,48 @@
+use std::cmp::max;
+
+use actix::{Actor, Handler, SyncContext};
+
+use near_performance_metrics_macros::perf;
+
+use crate::types::EdgeList;
+
+pub(crate) struct EdgeVerifier {}
+
+impl Actor for EdgeVerifier {
+    type Context = SyncContext<Self>;
+}
+
+impl Handler<EdgeList> for EdgeVerifier {
+    type Result = bool;
+
+    #[perf]
+    fn handle(&mut self, msg: EdgeList, _ctx: &mut Self::Context) -> Self::Result {
+        for edge in msg.edges {
+            let key = (edge.peer0.clone(), edge.peer1.clone());
+            if msg.edges_info_shared.lock().unwrap().get(&key).cloned().unwrap_or(0u64)
+                >= edge.nonce
+            {
+                continue;
+            }
+
+            #[cfg(feature = "adversarial")]
+            if !msg.adv_disable_edge_signature_verification && !edge.verify() {
+                return false;
+            }
+
+            #[cfg(not(feature = "adversarial"))]
+            if !edge.verify() {
+                return false;
+            }
+            {
+                let mut guard = msg.edges_info_shared.lock().unwrap();
```

Comment:
why do we need a mutex here?

Reply:
Yes, this is a data structure, that's shared between thread poll that `EdgeVerifier` uses. It should be safe.
I didn't change this code, just moved to another file.